Source/Core/DolphinQt/DiscordHandler.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+// Note using a ifdef around this class causes link issues with qt
```

Comment:
I think this was only due to missing ifdefs elsewhere. It compiles fine now, and this actually needs to be `ifdef`ed or else the linker fails.

Reply:
It still fails on my end but I did find a solution. I added ``-DUSE_DISCORD_PRESENCE`` into ``MocDefines`` in ``VSProps/QtCompile.props`` and it fixed the link issue.

Reply:
You mean adding that along with the `ifdef`?

Reply:
Oh, that is not acceptable then, don't push that. Maybe we should just ifdef out the member functions and leave the signals?

Reply:
Alternatively, we could ifdef all references to DiscordHandler, so the file doesn't need to be compiled at all if discord presence is disabled, which is probably a better solution anyways.